controllers/slicegateway/slicegateway.go

```diff
@@ -801,10 +841,21 @@ func (r *SliceGwReconciler) reconcileGatewayEndpoint(ctx context.Context, sliceG
 		return true, ctrl.Result{}, err
 	}
 	// endpoint already exists , check if sliceGatewayRemoteNodeIp is changed then update the endpoint
-	debugLog.Info("SliceGatewayRemoteNodeIP", "SliceGatewayRemoteNodeIP", sliceGw.Status.Config.SliceGatewayRemoteNodeIP)
-	if endpointFound.Subsets[0].Addresses[0].IP != sliceGw.Status.Config.SliceGatewayRemoteNodeIP {
-		debugLog.Info("Updating the Endpoint, since sliceGatewayRemoteNodeIp has changed", "from endpointFound", endpointFound.Subsets[0].Addresses[0].IP)
-		endpointFound.Subsets[0].Addresses[0].IP = sliceGw.Status.Config.SliceGatewayRemoteNodeIP
+	toUpdate := false
+	debugLog.Info("SliceGatewayRemoteNodeIP", "SliceGatewayRemoteNodeIP", sliceGw.Status.Config.SliceGatewayRemoteNodeIPs)
+	if len(sliceGw.Status.Config.SliceGatewayRemoteNodeIPs) > 1 {
```

Comment:
Not sure if I understand why the Addresses[0].IP is checked on line 847 and again on line 854 right after this block. Can we please loop through the endpointFound.Subsets[0].Addresses list and check if each element is present in SliceGatewayRemoteNodeIPs list? Please note, the SliceGatewayRemoteNodeIPs could contain more than two entries.

Reply:
This might be because of conflicts not fixed properly, will modify.